app/web/features/search/MapWrapper.tsx

```diff
@@ -0,0 +1,287 @@
+import maplibregl, { EventData, Map as MaplibreMap } from "maplibre-gl";
+import { addClusteredUsersToMap, layers } from "../search/users";
+import { reRenderUsersOnMap } from "features/search/users";
+import { useCallback, useEffect, useState } from "react";
+import ReplayIcon from "@material-ui/icons/Replay";
+import { Dispatch, SetStateAction } from "react";
+import { UserSearchRes } from "proto/search_pb";
+import TuneIcon from "@material-ui/icons/Tune";
+import { filterData } from "../search/users";
+import { InfiniteData } from "react-query";
+import makeStyles from "utils/makeStyles";
+import { usePrevious } from "utils/hooks";
+import { MutableRefObject } from "react";
+import Button from "components/Button";
+import { User } from "proto/api_pb";
+import Map from "components/Map";
+import { Point } from "geojson";
+import { SEARCH } from "i18n/namespaces";
+import { useTranslation } from "i18n";
+
+const useStyles = makeStyles((theme) => ({
+  searchHereGroup2: {
+    top: "20px",
+    // display: "flex",
+    left: "50%",
+    position: "relative",
+    zIndex: 10,
+  },
+  testContainer: {
+    position: "absolute",
+    left: "50%",
+    width: "100%",
+  },
+  testChildFromGoogle: {
+    width: "300px",
+    top: "30px",
+    display: "flex",
+    position: "relative",
+    left: "-50%",
+    fontSize: " 14px",
+    margin: "8px auto 0",
+    alignItems: "center",
+
+    height: "25px",
+    zIndex: 10,
+  },
+  buttonSearchSettings: {
+    borderRadius: "0 4px 4px 0",
+    "& span": {
+      margin: 0,
+    },
+  },
+  searchHereButton: {
+    borderRadius: "4px 0 0 4px",
+  },
+}));
+
+interface mapWrapperProps {
+  selectedResult:
+    | Pick<User.AsObject, "username" | "userId" | "lng" | "lat">
+    | undefined;
+  isLoading: boolean;
+  locationResult: any;
+  setLocationResult: Dispatch<SetStateAction<any>>;
+  results: InfiniteData<UserSearchRes.AsObject> | undefined;
+  setIsFiltersOpen: Dispatch<SetStateAction<boolean>>;
+  mapInitiallyLocated: boolean;
+  setSelectedResult: Dispatch<
+    SetStateAction<
+      Pick<User.AsObject, "username" | "userId" | "lng" | "lat"> | undefined
+    >
+  >;
+  map: MutableRefObject<MaplibreMap | undefined>;
+}
+
+export default function MapWrapper({
+  map,
+  selectedResult,
+  locationResult,
+  setLocationResult,
+  isLoading,
+  results,
+  mapInitiallyLocated,
+  setSelectedResult,
+  setIsFiltersOpen,
+}: mapWrapperProps) {
+  const { t } = useTranslation([SEARCH]);
+  const [areClustersLoaded, setAreClustersLoaded] = useState(false);
+  const previousResult = usePrevious(selectedResult);
+  const classes = useStyles();
+
+  /**
+   * User clicks on a user on map
+   */
+  const handleMapUserClick = useCallback(
+    (
+      ev: maplibregl.MapMouseEvent & {
+        features?: maplibregl.MapboxGeoJSONFeature[] | undefined;
```

Comment:
Won't this be undefined by default if it doesn't exist?

Reply:
the type you mean?

Reply:
Ah is that a type? Thought it was a function argument haha.

Might be worth it to make a separate type then as it's hard to see what's happening.

Reply:
🤔🤔🤔